feat(ai-gateway): Update AI Provider entity documentation (#5735)
Kong/developer.konghq.com: app/_ai_gateway_entities/ai-provider.md

  - text: AI Policy entity
    url: /ai-gateway/entities/ai-policy/
faqs:
  - q: What happens when I update an AI Provider's credentials?
    a: |
      {{site.ai_gateway}} propagates the credential change to every AI Model that references the
      AI Provider (by `name` or `id`). The next request through any of those AI Models uses the updated
      credentials.

  - q: How does an AI Model reference an AI Provider?
    a: |
      Set the `provider` field in each item of the [`targets`](/ai-gateway/entities/ai-model/#schema-aigateway-model-targets) array on the AI Model to the AI Provider's `name` or `id`.

  - q: Do AI Providers generate any runtime primitives on their own?
    a: |
      No. An AI Provider entity is a write-time template. Credentials and configuration only enter
      the runtime when an AI Model references the AI Provider; at that point, the AI Provider's values are

>       materialized into the underlying primitives generated for the AI Model.

Reviewer: ```suggestion
```
This is already in the body of the page si O think it can be removed

---

## What is an AI Provider?

> The AI Provider entity lets you securely store and manage credentials for connecting to upstream LLM services. Use AI Providers to store API keys for OpenAI, Azure, Bedrock, or any other LLM provider; centrally manage and rotate credentials across multiple AI Models; and enforce consistent authentication across your deployments.

Reviewer: ```suggestion
The AI Provider entity lets you securely store and manage credentials for connecting to upstream LLM services. Use AI Providers to: 
* Store API keys for OpenAI, Azure, Bedrock, or any other LLM provider
* Centrally manage and rotate credentials across multiple AI Models
* Enforce consistent authentication across your deployments
```

> Each AI Provider has a [`type`](#schema-aigateway-provider-type) that selects the upstream LLM service and configures provider-specific options. See the schema below for supported types, and the per-provider pages under [{{site.ai_gateway}} providers](/ai-gateway/ai-providers/) for provider-specific configuration and limitations.

Reviewer: ```suggestion
Each AI Provider has a [`type`](#schema-aigateway-provider-type) that selects the upstream LLM service and configures provider-specific options. See the [schema](#schema) below for supported types, and the per-provider pages under [{{site.ai_gateway}} providers](/ai-gateway/ai-providers/) for provider-specific configuration and limitations.
```

    fallback: "`basic`"
  - type: "`azure`"
    providers: "[Azure OpenAI](/ai-gateway/ai-providers/azure/)"
    approach: "Microsoft Entra ID via Managed Identity (recommended when running in Azure). For explicit credentials, provide client ID, secret, and tenant ID."
    fallback: "`basic`"
  - type: "`gcp`"
    providers: "[Gemini](/ai-gateway/ai-providers/gemini/), [Vertex AI](/ai-gateway/ai-providers/vertex/)"
    approach: "Google service accounts via environment auto-detection (service account JSON or Compute Engine metadata server). Custom metadata or OAuth token endpoints for restricted networks."
    fallback: "`basic`"
{% endtable %}

## Lifecycle

Creating an AI Provider stores the entity but doesn't generate any runtime primitives. AI Provider credentials enter the runtime only when an AI Model references the AI Provider. At that point, the credentials are materialized into the underlying primitives of the AI Model.

Updating an AI Provider re-materializes credentials into every AI Model that references it. The change takes effect on the next request through any referencing AI Model.

## AI Policies and AI Providers

> You cannot attach AI Policies directly to an AI Provider entity instance. Policies attach to [AI Models](/ai-gateway/entities/ai-model/), [AI Agents](/ai-gateway/entities/ai-agent/), [AI Consumers](/ai-gateway/entities/ai-consumer/), or [AI Consumer Groups](/ai-gateway/entities/ai-consumer-group/) to control security, rate limiting, guardrails, and observability.

Reviewer: ```suggestion
You can't attach [AI Policies](/ai-gateway/entities/ai-policy/) directly to an AI Provider entity instance. AI Policies attach to [AI Models](/ai-gateway/entities/ai-model/), [AI Agents](/ai-gateway/entities/ai-agent/), [AI Consumers](/ai-gateway/entities/ai-consumer/), or [AI Consumer Groups](/ai-gateway/entities/ai-consumer-group/) to control security, rate limiting, guardrails, and observability.
```

To apply an AI Policy across requests using a particular AI Provider, you can:
1. Set the policy to `global: true` to apply it to all resources in the gateway
2. Attach the same policy to each AI Model that references the AI Provider
3. Create an AI Consumer Group with the policy and control access to AI Models via ACLs

## Set up an AI Provider

> The following example creates an OpenAI Provider that authenticates with a single bearer-token header. A Model can then route to this Provider by setting the `provider` field in a `targets` array item to `my-openai-account` (or the Provider `id`).

Reviewer: ```suggestion
The following example creates an OpenAI Provider that authenticates with a single bearer-token header. An AI Model can then route to this AI Provider by setting the `provider` field in a `targets` array item to `my-openai-account` (or the AI Provider `id`).
```
